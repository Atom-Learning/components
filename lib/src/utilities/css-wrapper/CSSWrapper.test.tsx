import { render, screen } from '@testing-library/react'
import React from 'react'

import { CSSWrapper } from '.'

describe('CSSWrapper component', () => {
  it('renders if passed CSS blob', async () => {
    const { container } = render(
      <CSSWrapper css={{ m: 'auto', height: 100, width: 100 }}>
        <p>Example text</p>
      </CSSWrapper>
    )
    await container.querySelector('div')
    expect(container).toMatchSnapshot()
  })

  it("doesn't render if not passed CSS blob", async () => {
    const { container } = render(
      <CSSWrapper>
        <p>Example text</p>
      </CSSWrapper>
    )
    const wrapper = await container.querySelector('div')
    expect(wrapper).toBeNull()
    expect(container).toMatchSnapshot()
  })
})
