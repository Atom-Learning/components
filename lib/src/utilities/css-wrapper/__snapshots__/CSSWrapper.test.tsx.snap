// Vitest Snapshot v1

exports[`CSSWrapper component > doesn't render if not passed CSS blob 1`] = `
<div>
  <p>
    Example text
  </p>
</div>
`;

exports[`CSSWrapper component > renders if passed CSS blob 1`] = `
<div>
  <div
    class="c-PJLV c-PJLV-ikLAKdM-css"
  >
    <p>
      Example text
    </p>
  </div>
</div>
`;

exports[`CSSWrapper component doesn't render if not passed CSS blob 1`] = `
<div>
  <p>
    Example text
  </p>
</div>
`;

exports[`CSSWrapper component renders if passed CSS blob 1`] = `
@media  {
  .c-PJLV-ikLAKdM-css {
    margin: auto;
    height: 100px;
    width: 100px;
  }
}

<div>
  <div
    class="c-PJLV c-PJLV-ikLAKdM-css"
  >
    <p>
      Example text
    </p>
  </div>
</div>
`;
